keras_contrib/applications/densenet.py

```diff
@@ -501,10 +527,24 @@ def __create_dense_net(nb_classes, img_input, include_top, depth=40, nb_dense_bl
     x = BatchNormalization(axis=concat_axis, gamma_regularizer=l2(weight_decay),
                            beta_regularizer=l2(weight_decay))(x)
     x = Activation('relu')(x)
-    x = GlobalAveragePooling2D()(x)
+    if pooling is not None:
+        x = GlobalAveragePooling2D()(x)
 
-    if include_top:
-        x = Dense(nb_classes, activation='softmax', kernel_regularizer=l2(weight_decay), bias_regularizer=l2(weight_decay))(x)
+    if include_top and pooling is not None:
```

Comment:
If the argument states to include_top and to pool, you are adding a Dense layer with no pooling. The code does not match your intention. Either use some other parameter to decide to add a Dense layer or add another argument to DenseNet to allow this behavior. 

Logically, it is counter intuitive to use a dense layer which obviously doesn't pool when the pool argument is not None and clearly states some form of pooling should occur. Such confusion gives rise to wrong models being built.